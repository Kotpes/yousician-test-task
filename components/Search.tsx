import React, { ChangeEvent, FunctionComponent } from 'react'

import styles from './Search.module.css'

interface Props {
  searchValue: string;
  updateSearchValue: (event: ChangeEvent<HTMLInputElement>) => void
}

const Search: FunctionComponent<Props> = (props: Props) => {
  const { searchValue, updateSearchValue } = props
  return (
    <section className={styles.searchContainer}>
      <input
        value={searchValue}
        type='text'
        name='songs-search'
        id='songsSearch'
        placeholder='Search for songs by artist or title'
        className={styles.searchInput}
        onChange={updateSearchValue}
      />
      <img
        src='/icons/search.svg'
        alt='search icon'
        className={styles.searchIcon}
      />
    </section>
  )
}


export default Search

