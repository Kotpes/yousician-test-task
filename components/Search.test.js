import { create } from 'react-test-renderer';
import { fireEvent, render, screen } from '@testing-library/react';
import Search from './Search';

const updateSearchValue = jest.fn();

const props = {
  searchValue: 'Jackson 5',
  updateSearchValue,
};

describe('Search', () => {
  test('it matches snapshots', () => {
    const component = create(<Search {...props} />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('it has correct initial value', () => {
    const component = create(<Search {...props} />);
    const instance = component.root;
    const inputElement = instance.findByType('input');

    expect(inputElement.props.value).toEqual('Jackson 5');
  });

  test('it has correct value after change', () => {
    render(<Search {...props} />);
    const inputField = screen.getByPlaceholderText(
      'Search for songs by artist or title'
    );
    fireEvent.change(inputField, { target: { value: 'Beatles' } });
    expect(updateSearchValue).toHaveBeenCalled();
  });
});
