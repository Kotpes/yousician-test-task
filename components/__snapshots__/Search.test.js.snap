// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search it matches snapshots 1`] = `
<section
  className="searchContainer"
>
  <input
    className="searchInput"
    id="songsSearch"
    name="songs-search"
    onChange={[MockFunction]}
    placeholder="Search for songs by artist or title"
    type="text"
    value="Jackson 5"
  />
  <img
    alt="search icon"
    className="searchIcon"
    src="/icons/search.svg"
  />
</section>
`;
